service/src/main/java/bio/terra/workspace/service/workspace/model/Workspace.java

```diff
@@ -191,4 +192,43 @@ public Workspace build() {
           workspaceStage);
     }
   }
+
+  /**
+   * If requester only has discoverer role, they can only see a subset of workspace. They cannot
+   * see:
+   *
+   * <ul>
+   *   <li>Workspace description
+   *   <li>Spend profile
+   *   <li>Can see special type, short description and version properties, and no other properties
+   * </ul>
+   */
+  public static Workspace stripWorkspaceForRequesterWithOnlyDiscovererRole(
```

Comment:
I thought you already added this function for the "list" PR?

Reply:
In List PR, I added this to `WorkspaceService.java`. In this PR, I moved to `model/Workspace.java`, since controller also needs it.